### PR TITLE
Fix rare cases where CNAME and MX only contain a dot

### LINKED ISSUE #23: Dot (`.`) in the value of MX record causes the module to freak out
Hi,

we currently have the `.` set as the `MX` records for some of our domains.

```
? ''
: - octodns:
      cloudflare:
        proxied: true
    ttl: 300
    type: ALIAS
    value: some.domain.
  - ttl: 86400
    type: TXT
    value: v=spf1 -all
  - ttl: 86400
    type: MX
    value:
      exchange: .
      preference: 0
```

While retrieving the current values from Cloudflare, octoDNS screams that `..` is not a valid FQDN:
```
2022-07-11T12:53:24  [4620371456] WARNING Record Invalid record testing.domain.
  - Invalid MX exchange ".." is not a valid FQDN.
```

While the Cloudflare API does not usually return records with the trailing dot, there's an exception: if the record only contains a dot and nothing else, the API responds with just the dot:
```
    {
(...)
      "zone_name": "testing.domain",
      "name": "testing.domain",
      "type": "CNAME",
      "content": "some.domain",
(...)
    },
    {
(...)
      "zone_name": "testing.domain",
      "name": "testing.domain",
      "type": "MX",
      "content": ".",
(...)
    },
```

It looks like this line is the culprit:
https://github.com/octodns/octodns-cloudflare/blob/3e8fe9b8bd808cdc7073e27867643f606be1b2bc/octodns_cloudflare/__init__.py#L256

Funny enough, someone already hit this problem with the `SRV` records, as the value of `target` for the `SRV` record is checked for a dot:
https://github.com/octodns/octodns-cloudflare/blob/3e8fe9b8bd808cdc7073e27867643f606be1b2bc/octodns_cloudflare/__init__.py#L275

Apparently there should be checks for `'.'` for all the other records where dot is appended (`CNAME`, `MX` and maybe even `NS`, although I can't imagine a case where someone would place a dot into the value of a `NS` record).

### PR DESCRIPTION
**Pull request by zajdee**
Fixes #23

### CODE REVIEW

File: octodns_cloudflare/__init__.py [hunk 1]
```diff
@@ -217,7 +217,7 @@ def _data_for_CNAME(self, _type, records):
         return {
             'ttl': self._ttl_data(only['ttl']),
             'type': _type,
-            'value': f'{only["content"]}.',
+            'value': f'{only["content"]}.' if only['content'] != "." else ".",
```
Comment (ross) [inline octodns_cloudflare/__init__.py:220]:
```suggestion
            'value': f'{only["content"]}.' if only['content'] != '.' else '.',
```

File: octodns_cloudflare/__init__.py [hunk 2]
```diff
@@ -253,7 +253,12 @@ def _data_for_MX(self, _type, records):
         values = []
         for r in records:
             values.append(
-                {'preference': r['priority'], 'exchange': f'{r["content"]}.'}
+                {
+                    'preference': r['priority'],
+                    'exchange': f'{r["content"]}.'
+                    if r['content'] != "."
+                    else ".",
```
Comment (ross) [inline octodns_cloudflare/__init__.py:260]:
```suggestion
                    if r['content'] != '.'
                    else '.',
```

**Review by ross — APPROVED** [review]
octoDNS coding style prefers `'` over `"`.

**Comment by ross** [comment]
Thanks

**Comment by zajdee** [comment]
Hi @ross,
thank you for merging this PR.

The quotes were actually a controversial point in a discussion with my colleagues. From the two options I have picked `"."` to be in line with the SRV record processing. Next time I'll know. 😀

**Comment by ross** [comment]
> The quotes were actually a controversial point in a discussion with my colleagues. From the two options I have picked `"."` to be in line with the SRV record processing. Next time I'll know. 😀

Heh. Not a big deal either way. It does look like this module uses `"` in places anyway. Something I wish `pycodestyle` (in the past) or `black` (now) dealt with one way or another so that humans didn't have to think about it :grin:

**Pull request merged**